src/arch/MovieTexture/MovieTexture_FFMpeg.cpp

```diff
@@ -217,6 +218,103 @@ float MovieDecoder_FFMpeg::GetFrameDuration() const
 	return m_fLastFrameDelay;
 }
 
+int MovieDecoder_FFMpeg::SendPacketToBuffer()
+{
+	if (cancel) return -2;
+	if (m_iEOF > 0) return 0;
+
+	for (;;)
+	{
+		int ret = avcodec::av_read_frame(m_fctx, &m_FrameBuffer.back().packet);
+		/* XXX: why is avformat returning AVERROR_NOMEM on EOF? */
+		if (ret < 0)
+		{
+			/* EOF. */
+			m_iEOF = 1;
+			m_FrameBuffer.pop_back(); // Don't display an EoF frame.
+			// If we had to approximate the number of frames, set the actual
+			// total number of frames. This is benign even if we did have an
+			// accurate frame count at the start.
+			m_totalFrames = m_FrameBuffer.size();
+			return 0;
+		}
+
+		if (m_FrameBuffer.back().packet.stream_index == m_pStream->index)
+		{
+			m_iCurrentPacketOffset = 0;
+			return 1;
+		}
+		/* It's not for the video stream; ignore it. */
+		avcodec::av_packet_unref(&m_FrameBuffer.back().packet);
+	}
+}
+
+int MovieDecoder_FFMpeg::DecodePacketInBuffer() {
+	if (cancel) return -2;
+	if (m_iEOF == 0 && m_iCurrentPacketOffset == -1)
+		return 0; /* no packet */
+
+	while (m_iEOF == 0 && m_iCurrentPacketOffset <= m_FrameBuffer.back().packet.size)
+	{
+		/* If we have no data on the first frame, just return EOF; passing an empty packet
+		 * to avcodec_decode_video in this case is crashing it.  However, passing an empty
+		 * packet is normal with B-frames, to flush.  This may be unnecessary in newer
+		 * versions of avcodec, but I'm waiting until a new stable release to upgrade. */
+		if (m_FrameBuffer.back().packet.size == 0 && firstFrame)
+			return 0; /* eof */
+
+		int iGotFrame;
```

Comment:
Let's declare and initialize on the same line below.

Reply:
Done.